### ISSUE TITLE
Igx-grid cannot use arrowkeys etc inside input / dropdown elements in cell

### ISSUE THREAD

**Issue by VirtualisProject** [IgniteUI/igniteui-angular#5819]
## Question  
I have an issue with editing values inside inputs and drops-downs if they are inside a cell in the igx grid. I assume this has to do with cell focus, and would like to investigate if there is a way to accomplish what i am trying to do, while still using the igx-grid.

I've attached a stackblitz which displays my issue:
https://stackblitz.com/edit/grid-matselect-sample-ej5yty

1. Editing the value inside the input fields with either arrow keys, space etc either makes the input field loose it's focus, or does nothing at all.

2. Toggling between the different options in the mat-select makes the mat-select loose focus to a cell based in the arrow key pressed.

I've tried setting the IgxFocus property to the input field, but that does not seem to accomplish anything. 

PS: We are not, and cannot use igx-grid built-in editing because we have functionality that was implemented before igx-grid received support for in-line editing. 

Would REALLY appriciate a fast answer, so we can move forward or make this work as we have a release coming very soon :) 

Thanks! 

 * igniteui-angular version: 7.3.4
 * browser: Google Chrome

**Comment by zdrawku** [comment]
Hey @VirtualisProject, apologies for not taking the time to answer your questions sooner! I hope you will like the new version that we've released last week, it took most of our time and it’s worth it 🎉

The behavior that you've described is due to [keyboard navigation](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/keyboard_navigation.html#key-combinations) functionality, it uses the focus to navigate through the cells and other grid elements - in this particular scenario this causes the editing malfunction of the arrow keys and space key. 

As you assumed, my suggestion would be to use our build-in editing and with that said I can assure you that if you need any help with migrating to it - I can provide it. For me, it is important to try and start using the variety of new functionalities that we've released with the latest 8.1 and 8.2 versions of our product, and with that said I believe you will very much like our row-editing feature.

Another approach that I can recommend is to wrap your input elements with a container and stop the `keydown` event propagation from there. [The sample](https://stackblitz.com/edit/grid-stop-propagation-of-arrows) is updated accordingly. This approach is clean and efficient, you would be able to use the arrow and space keys for editing and it wont stop the `tab` navigation capabilities of the Grid, I hope you will like it.


```ts
<igx-column field="name" [dataType]="'string'" [width]="'33%'">
        <ng-template igxCell let-cell="cell">
            <div (keydown)="$event.stopPropagation()"><input matInput igxFocus [igxFocus]="cell.focused" [ngModel]="cell.value" /></div>
        </ng-template>   
      </igx-column>
```

**Comment by VirtualisProject** [comment]
Hey @zdrawku, 

Thanks for getting back to me, this solved my issue :) 

We'll definatly look into the 8.1 /8.2 version and see if we have time to migrate to the build-in editing :)
